### PR TITLE
Cleanup implementation of macros

### PR DESCRIPTION
**Pull request by vtjnash**
Fixes hygiene bug and global misuse.

### CODE REVIEW

**Comment by vtjnash** [comment]
bump?

File: src/LazilyInitializedFields.jl [hunk 1]
```diff
@@ -292,9 +288,8 @@ function lazy_struct(expr, mod)
     expr.args[1] = true # make mutable
     lazyfield = QuoteNode[]
     for (i, arg) in enumerate(body.args)
-        if arg isa Expr && arg.head === :macrocall && arg.args[1] === Symbol("@lazy")
-            body.args[i] = macroexpand(mod, arg)
-            name = body.args[i].args[1]
+        if arg isa Expr && arg.head === :macrocall && length(arg.args) == 3 && arg.args[1] === Symbol("@lazy")
+            name, body.args[i] = lazy_field(mod, arg.args[3])
```
Comment (KristofferC) [inline src/LazilyInitializedFields.jl]: Looks like this is trying to call a function that doesn't exist (two arg `lazy_field`)?

Comment (vtjnash) [reply src/LazilyInitializedFields.jl]: oops, yep, does not seem to need `mod` anymore
```suggestion
            name, body.args[i] = lazy_field(arg.args[3])
```

**Comment by vtjnash** [comment]
Merge?

**Pull request merged**